sdk/digitaltwins/Azure.DigitalTwins.Core/tests/AdtQueryBuilderTests/WhereLogicTests.cs

```diff
@@ -19,6 +19,26 @@ public void WhereLogic_Comparison()
                 .Be("Temperature = 5");
         }
 
+        [Test]
+        public void WhereLogic_NumberWithStringComparison()
+        {
+            var query = new WhereLogic(null);
+            query.Compare("Temperature", QueryComparisonOperator.Equal, "'5'");
+            query.GetQueryText()
+                .Should()
+                .Be("Temperature = 5");
```

Comment:
interesting, how else would the user tell you I want this to be treated as a string?

Reply:
Interesting case. I didn't consider that, let me think about that now